swift/Samples/TabBarContainer/Sources/TabBarScreenViewController.swift

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Workflow
+import WorkflowUI
+
+internal final class TabBarScreenContainerViewController<Content: Screen>: ScreenViewController<TabBarScreen<Content>>, UITabBarDelegate {
+    private var containerView: ContainerView
+    private var contentViewController: DescribedViewController
+    
+    private let tabBar: UITabBar
+    
+    required init(screen: TabBarScreen<Content>, environment: ViewEnvironment) {
+        self.containerView = ContainerView()
+        
+        self.contentViewController = DescribedViewController(
+            screen: screen.currentScreen,
+            environment: environment
+        )
+        
+        self.tabBar = UITabBar()
+        
+        super.init(screen: screen, environment: environment)
+        
+        self.tabBar.delegate = self
+    }
+
+    override internal func screenDidChange(from previousScreen: TabBarScreen<Content>, previousEnvironment: ViewEnvironment) {
+        update()
+    }
+
+    private func update() {
+        contentViewController.update(screen: screen.currentScreen, environment: environment)
+        
+        var selectedTabBarItem: UITabBarItem?
+        
+        let tabBarItems = screen.barItems.enumerated().map { (index, barItem) -> UITabBarItem in
+            let tabBarItem = UITabBarItem(
+                title: barItem.title,
+                image: barItem.image,
+                selectedImage: barItem.selectedImage
+            )
+            
+            tabBarItem.badgeValue = barItem.badge.stringValue
+            
+            if index == screen.selectedIndex {
+                selectedTabBarItem = tabBarItem
+            }
+            
+            return tabBarItem
+        }
+        
+        tabBar.setItems(tabBarItems, animated: false)
+        tabBar.selectedItem = selectedTabBarItem
+    }
+
+    override internal func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(containerView)
```

Comment:
Any reason we shouldn't use `UITabBarController` here?

Reply:
I've seen two internal implementations that use BluePrint to render the tab bar *and* only accept the currently displayed screen. 

This allows greater control of what the UI rendering / view behavior looks like. For this sample, I didn't want to use Blueprint, so decided to use a simple UITabBar. 

The question of holding onto an array of multiple screens is a good one and should be discussed in more detail.  I'd like a few more folks to weigh their opinions on this PR before we circle up as a group to decide.